dotCMS/src/integration-test/java/com/dotcms/datagen/AppDescriptorDataGen.java

```diff
@@ -0,0 +1,166 @@
+import com.dotcms.security.apps.Type;
+import com.dotmarketing.util.Logger;
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.collect.ImmutableSortedMap;
+import com.google.common.collect.Ordering;
+import com.liferay.util.StringPool;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.SortedMap;
+
+public class AppDescriptorDataGen extends AbstractDataGen<AppDescriptor> {
+
+    private static final ObjectMapper ymlMapper = new ObjectMapper(new YAMLFactory())
+            //.enable(SerializationFeature.INDENT_OUTPUT)
+            .setVisibility(PropertyAccessor.FIELD, Visibility.ANY)
+            .findAndRegisterModules();
+
+    private String key = String.format("app_%d", System.currentTimeMillis());
+    private String name = key;
+    private String fileName = String.format("%s.yml", key);
+    private  String description = "";
+    private boolean allowExtraParameters = false;
+    private String iconUrl = "none";
+
+    private SortedMap<String, ParamDescriptor> paramMap;
+
+    private SortedMap<String, ParamDescriptor> paramMap(){
+       if(null == paramMap){
+          paramMap = builder.build();
+       }
+       return paramMap;
+    }
+
+    private ImmutableSortedMap.Builder<String, ParamDescriptor> builder = new ImmutableSortedMap.Builder<>(Ordering.natural());
+
+    @Override
+    public AppDescriptor next() {
+        return new AppDescriptor(key, name, description, iconUrl, allowExtraParameters, paramMap());
+    }
+
+    @Override
+    public AppDescriptor persist(AppDescriptor object) {
+        try (InputStream in = persistDescriptor(object)) {
+            Logger.debug(AppDescriptorDataGen.class, in.toString());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return object;
+    }
+
+    private InputStream persistDescriptor(final AppDescriptor object) throws IOException {
+        String basePath = System.getProperty("java.io.tmpdir");
+        basePath = Paths.get(basePath).normalize().toString();
+        final File file = new File(basePath, fileName);
+        ymlMapper.writeValue(file, object);
+        return Files.newInputStream(Paths.get(file.getPath()));
+    }
+
+    public InputStream nextPersistedDescriptor() throws IOException {
+        return persistDescriptor(next());
+    }
+
+    public void destroy() throws IOException {
+        String basePath = System.getProperty("java.io.tmpdir");
+        basePath = Paths.get(basePath).normalize().toString();
+        final File file = new File(basePath, fileName);
+        Files.delete(Paths.get(file.getPath()));
+    }
+
+    @SuppressWarnings("unused")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "unused" appears 15 times in this file; the first occurrence is on line 80](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5358201)